osu.Game/Rulesets/UI/Playfield.cs

```diff
@@ -443,6 +452,25 @@ public DrawableSamplePool(ISampleInfo sampleInfo, int initialSize, int? maximumS
 
         #endregion
 
+        private void onNewResult(DrawableHitObject drawable, JudgementResult result)
+        {
+            // Not using result.TimeAbsolute because that might change and also there is a potential precision issue.
+            judgementResults.Push(new JudgementResultEntry(Time.Current, drawable.Entry.AsNonNull(), result));
+
+            NewResult?.Invoke(drawable, result);
+        }
+
+        private void revertResult(JudgementResultEntry entry)
+        {
+            var result = entry.Result;
+            RevertResult?.Invoke(result);
+
+            result.TimeOffset = 0;
+            result.Type = HitResult.None;
+
+            entry.HitObjectEntry.OnRevertResult();
```

Comment:
The fact that `Playfield.RevertResult` event subscribers see `result` with everything populated, but `HitObjectLifetimeEntry.RevertResult` subscirber (and therefore, consequently, `DrawableHitObject.RevertResult` subscribers) are seeing the result with time offset and type cleared scares me a lot. My confidence that this is correct is basically zero unless I go and chase down all event pathways. Is this preserving existing behaviour?

Reply:
Good catch. This was not intentional, no.